### PR TITLE
Added List manager, steps and features.

### PR DESCRIPTION
**Pull request by jorgeposterla**
```
###**These tests depend on the PR of Nicolás and Daniel and cannot be Merged**
Feature: Trello API Lists # tests/api/features/lists.feature:1
  As a user
  I want "Lists" API endpoints
  In order to manipulate lists
  Scenario: GET all lists in a board                      # tests/api/features/lists.feature:6
    Given I created a board                               # tests/api/features/steps/lists_steps.py:10
    And I created a list on the board with name "list 22" # tests/api/features/steps/lists_steps.py:14
    When I send a "GET" request to "boards/{board}/lists" # tests/api/features/steps/common_steps.py:8
    Then I receive a list with at least "1" "list"        # tests/api/features/steps/common_steps.py:14
    And the status code is "200"                          # tests/api/features/steps/common_steps.py:23

  Scenario: GET a list by id                              # tests/api/features/lists.feature:13
    Given I created a board                               # tests/api/features/steps/lists_steps.py:10
    And I created a list on the board with name "list 22" # tests/api/features/steps/lists_steps.py:14
    When I send a "GET" request to "/lists/{list}"        # tests/api/features/steps/common_steps.py:8
    Then I receive a response with the "list" id          # tests/api/features/steps/common_steps.py:18
    And the status code is "200"                          # tests/api/features/steps/common_steps.py:23

  Scenario: POST a new list to a board                 # tests/api/features/lists.feature:20
    Given I created a board                            # tests/api/features/steps/lists_steps.py:10
    When I send a "POST" request to "/lists"           # tests/api/features/steps/common_steps.py:8
      | Key     | Value      |
      | name    | list 1     |
      | idBoard | {id}:board |
    Then I receive a response with the "list" schema   # tests/api/features/steps/common_steps.py:27
    And the status code is "200"                       # tests/api/features/steps/common_steps.py:23
    When I send a "GET" request to "/lists/{response}" # tests/api/features/steps/common_steps.py:8
    Then I check the data is                           # tests/api/features/steps/lists_steps.py:19
      | Key     | Value      |
      | name    | list 1     |
      | idBoard | {id}:board |
```

### CODE REVIEW

File: main/trello/api/lists_manager.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+"""
+class ListsManager(RESTBaseManager):
+    """Class which can be used to manage Trello Lists through the API"""
+
+    def __init__(self, request_method=None):
+        super().__init__(request_method)
+        """ Construct the necessary attributes for the ListsManager
+
+        :param method: obj  RequestManager object which is used to handle the API requests
+        """
```
Comment (mefemefe) [inline main/trello/api/lists_manager.py:18]: The doc string needs to be right below the signature:
```suggestion
        """ Construct the necessary attributes for the ListsManager
        :param method: obj  RequestManager object which is used to handle the API requests
        """
        super().__init__(request_method)
```

File: main/trello/api/lists_manager.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    
+    def create_list(self, name, idBoard):
+        """ Create a List in a Board
+
+        :param name: str   Name of the List
+        :param idBoard:  str   ID of the Board to which the List belongs
+        :return: Tuple that contains the status code and the response.
+        """
+        endpoint = f"lists/?name={name}&idBoard={idBoard}"
+        payload = { "name": name}
```
Comment (mefemefe) [inline main/trello/api/lists_manager.py:28]: Probably a nitpick, but sending a parameter ('name') both as a query parameter AND inside the payload seems redundant and unnecessary.

File: tests/api/features/steps/lists_steps.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+def step_impl(context, list_name):
+    if not getattr(context, 'list'):
+        context.status_code, context.list = manager.create_list(list_name, context.board["id"])
+
+@step('I check the data is')
+def step_impl(context):
+    expected_json = fill_payload(context, payload={})
+    for key, value in expected_json.items():
+        assert value==context.response[key], \
+        'expected value for {key} is {value},but actual is {context.response[key]}'
```
Comment (mefemefe) [inline tests/api/features/steps/lists_steps.py:24]: If the idea is to show the actual "key" and "value", then the string is missing a formatter:
```suggestion
        f'expected value for {key} is {value}, but actual is {context.response[key]}'
```

File: main/trello/api/lists_manager.py (hunk 2 above)
Comment (jorgeposterla) [reply main/trello/api/lists_manager.py]: Resolved with the changes suggested.

**Review by nicolasvallejo13 — APPROVED** [review]

**Pull request closed without merging**